mirrord-layer/src/lib.rs

```diff
@@ -222,6 +180,20 @@ where
 
                 self.codec.send(request).await.unwrap();
             }
+            HookMessage::TcpOutgoing(message) => self
+                .tcp_outgoing_handler
+                .handle_hook_message(message, &mut self.codec)
+                .await
+                .unwrap(),
+            HookMessage::Exit(message) => {
+                trace!("HookMessage::Exit");
+
+                let request = ClientMessage::ExitRequest;
+
+                self.codec.send(request).await.unwrap();
+
+                message.hook_channel_tx.send(Ok(())).unwrap();
```

Comment:
break loop for graceful exit here

Reply:
ah we need to send a message when we close.. ermm.. we could call exit on our own (bypassing the hook becuase of our bypass) when loop ends..?

Reply:
Both ways are pretty valid (imo), so:

1. Should I change it now?
2. Keep as-is is fine?
3. Keep as-is, but add a `TODO`?

Reply:
I'm okay with 3, just add a comment.